src/diffusers/models/unet_2d_condition.py

```diff
@@ -528,6 +528,30 @@ def fn_recursive_add_processors(name: str, module: torch.nn.Module, processors:
 
         return processors
 
+    @property
+    def attn_processors_state_dict(self) -> Dict[str, torch.tensor]:
+        r"""
+        Returns:
+            a state dict containing just the attention processor parameters.
+        """
+        # set recursively
+        processors = {}
+
+        def fn_recursive_add_processors(name: str, module: torch.nn.Module, processors: Dict[str, AttentionProcessor]):
+            if hasattr(module, "set_processor"):
+                for processor_key, processor_parameter in module.processor.state_dict().items():
+                    processors[f"{name}.processor.{processor_key}"] = processor_parameter
+
+            for sub_name, child in module.named_children():
+                fn_recursive_add_processors(f"{name}.{sub_name}", child, processors)
+
+            return processors
+
+        for name, module in self.named_children():
+            fn_recursive_add_processors(name, module, processors)
+
+        return processors
+
```

Comment:
counter part to text_encoder_lora_state_dict. Make it very clear how we get the subset of the state dict